### PR TITLE
(CV) Fix rake aborted bug when moab does not have correct file or dir

### PR DESCRIPTION
**Pull request by SaravShah**
When a moab does not have the correct file or directory, the rake task would throw an error. 
```
be rake cv_all_endpoints  
rake aborted!
Errno::ENOENT: No such file or directory @ rb_sysopen - spec/fixtures/storage_root02/moab_storage_trunk/yy/000/yy/0000/yy000yy0000/v0001/manifests/signatureCatalog.xml
```

connects #618

### CODE REVIEW

File: spec/services/checksum_validator_spec.rb [hunk 1]
```diff
@@ -315,5 +315,22 @@
...
+
+    it 'rescues if the file or directory does not exist' do
+      druid = 'yy000yy0000'
+      endpoint_name = 'fixture_sr2'
+      endpoint = Endpoint.find_by(endpoint_name: endpoint_name)
+      po = PreservedObject.find_by(druid: druid)
+      pres_copy = PreservedCopy.find_by(preserved_object: po, endpoint: endpoint)
+      cv = described_class.new(pres_copy, endpoint_name)
+      results = instance_double(AuditResults, report_results: nil, :check_name= => nil)
+      # allow(AuditResults).to receive(:new).and_return(results)
```
Comment (ndushay) [inline spec/services/checksum_validator_spec.rb]: whoops!  don't need this comment.

File: spec/services/checksum_validator_spec.rb [hunk 2]
```diff
@@ -315,5 +315,22 @@
       expect(cv).to receive(:flag_unexpected_data_files)
       cv.validate_signature_catalog
     end
+
+    it 'rescues if the file or directory does not exist' do
```
Comment (ndushay) [inline spec/services/checksum_validator_spec.rb]: `adds error code and continues executing if a file or directory does not exist` ??

File: app/services/checksum_validator.rb [hunk 3]
```diff
@@ -137,8 +137,22 @@ def latest_signature_catalog_path
...
+    @latest_signature_catalog_entries ||= begin
+      if latest_moab_version.signature_catalog
+        latest_moab_version.signature_catalog.entries
+      else
+        absent_from_moab_data = { signature_catalog_path: latest_moab_version.signature_catalog }
+        checksum_results.add_result(AuditResults::SIGNATURE_CATALOG_NOT_IN_MOAB, absent_from_moab_data)
+        []
+      end
+    end
+  rescue Errno::ENOENT, NoMethodError
```
Comment (ndushay) [inline app/services/checksum_validator.rb]: maybe add comment such as `we get here when ...`

File: spec/services/checksum_validator_spec.rb (hunk 1 above)
Comment (tallenaz) [inline spec/services/checksum_validator_spec.rb]: Seems like this comment could disappear?

File: app/services/checksum_validator.rb [hunk 4]
```diff
@@ -137,8 +137,22 @@ def latest_signature_catalog_path
     @latest_signature_catalog_path ||= latest_moab_version.version_pathname.join(MANIFESTS, SIGNATURE_XML).to_s
   end
 
+  # shameless green implementation
   def latest_signature_catalog_entries
-    @latest_signature_catalog_entries ||= latest_moab_version.signature_catalog.entries
+    @latest_signature_catalog_entries ||= begin
+      if latest_moab_version.signature_catalog
+        latest_moab_version.signature_catalog.entries
```
Comment (tallenaz) [inline app/services/checksum_validator.rb]: Nice catch!

**Pull request merged**